### ISSUE TITLE
Not able to apply kotlin-noarg plugin in multi-project setup

### ISSUE THREAD

**Issue by xmlking** [gradle/kotlin-dsl#370]
I have multi project gradle setup. Trying to define `org.jetbrains.kotlin.plugin.noarg` plugin in root project `(apply false)` and apply and use it in sub-project. But I am getting following error:
```
build.gradle.kts:18:1: Unresolved reference: noArg
build.gradle.kts:19:5: Unresolved reference: annotation
```

Root project : build.gradle.kts
```groovy
plugins {
    val kotlinVersion = "1.1.2-2"
    base
    id("org.jetbrains.kotlin.jvm") version kotlinVersion apply false
    id("org.jetbrains.kotlin.plugin.spring") version kotlinVersion apply false
    id("org.jetbrains.kotlin.plugin.noarg") version kotlinVersion apply false
}
```
sub project : build.gradle.kts
```groovy 
apply {
    plugin("org.jetbrains.kotlin.jvm")
    plugin("org.jetbrains.kotlin.plugin.spring")
    plugin("org.jetbrains.kotlin.plugin.noarg")
}
noArg {
    annotation("org.springframework.data.mongodb.core.mapping.Document")
}
```
Example project to reproduce this issue 
https://github.com/xmlking/reactive-apps

Environment :
```bash
./gradlew -v

------------------------------------------------------------
Gradle 4.0-20170518042627+0000
------------------------------------------------------------

Build time:   2017-05-18 04:26:27 UTC
Revision:     7d181c4bec00ddf26f98dc898c341647f13f968e

Groovy:       2.4.11
Ant:          Apache Ant(TM) version 1.9.6 compiled on June 29 2015
JVM:          1.8.0_101 (Oracle Corporation 25.101-b13)
OS:           Mac OS X 10.12.5 x86_64
```

**Comment by weitjong** [comment]
I faced the same issue and started googling for the answer. That's how I found this issue log here. At first I thought not a good sign as there was no answer yet and that I need to come up with the answer myself. Fortunately,  my search result included a link to one of the sample project in the GSK which has enlightened me to figure out the answer to our issue is not far from number 42. Below is what I come up with. Not sure if there is a better way but it solves my issue.

At this at the top of the script.
```
import org.jetbrains.kotlin.noarg.gradle.NoArgExtension
```
And this after applying the plugin.
```
configure<NoArgExtension> {
    annotation("your.annotation.Class")
}
```
